iceoryx_hoofs/source/posix_wrapper/scheduler.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/posix_wrapper/scheduler.hpp"
+#include "iceoryx_hoofs/cxx/requires.hpp"
+#include "iceoryx_hoofs/internal/log/hoofs_logging.hpp"
+#include "iceoryx_hoofs/posix_wrapper/posix_call.hpp"
+
+namespace iox
+{
+namespace posix
+{
+int32_t getSchedulerPriorityMinimum(const Scheduler scheduler) noexcept
+{
+    auto result = posixCall(sched_get_priority_min)(static_cast<int>(scheduler)).failureReturnValue(-1).evaluate();
+    if (result.has_error())
+    {
+        LogError() << "The \"sched_get_priority_min\" should never fail. This can only be caused by an internal logic "
+                      "error or a non posix compliant system.";
+
+        // NOLINTJUSTIFICATION Required to provide an error message to the user
+        // NOLINTNEXTLINE(hicpp-no-array-decay,cppcoreguidelines-pro-bounds-array-to-pointer-decay)
```

Comment:
I think this is now suppressed in `requires.hpp` and not needed here.

Reply:
no it is still required.

Reply:
Are you sure? I just wonder why the suppression is not needed in e.g. `thread.cpp` and `system_configuration.cpp`.

Reply:
My IDE is still showing me a warning when I remove those lines.

Reply:
My IDE not, but this is something for #1615.